src/components/modals/PrivacySettingsModal.vue

```diff
@@ -142,12 +192,23 @@ export default {
       }
     },
 
-    addWhitelistedAddress() {
+    toggleWhitelistField() {
 
-      const addressToAdd = this.newWhitelistedAddress
+      if (!config.supportEmailAccounts) {
+        this.showEthereumAddressField = true
+        return
+      }
+
+      this.toEthAddress = null
+      this.toEmailAddress = null
+      this.showEthereumAddressField = !this.showEthereumAddressField
+    },
+
+    addWhitelistedAddress(addressToAdd = this.newWhitelistedAddress) {
 
       if (
         addressToAdd !== null &&
+        this.instance.utils.isAddress(addressToAdd) &&
```

Comment:
Does this make the `!== null` comparison redundant?

Reply:
Yeah it does. I was actually gonna remove the null check for this reason, but I decided to leave it since we do the same null check in `addWhitelistedEmail()`. I suppose I can remove it though.